fetchbuild.sh

```diff
@@ -38,16 +43,67 @@ elif [ "$target_host" = "aarch64-linux-android" ]; then
     NDKARCH=arm64
 fi
 
-NDKV=19
-if [ "$bits" = "64" ]; then
-    NDKV=21
-fi
-
 num_jobs=4
 if [ -f /proc/cpuinfo ]; then
     num_jobs=$(grep ^processor /proc/cpuinfo | wc -l)
 fi
 
+## build cln first
+
+# sqlite
+unpackdep https://www.sqlite.org/2018/sqlite-autoconf-3260000.tar.gz 5daa6a3fb7d1e8c767cd59c4ded8da6e4b00c61d3b466d0685e35c4dd6d7bf5d
+cd sqlite-autoconf-3260000
+./configure --enable-static --disable-readline --disable-threadsafe --host=${target_host} CC=$CC --prefix=${LNBUILDROOT}
+make -j $num_jobs
+make install
+cd ..
+rm -rf sqlite-autoconf-3260000
+
+# gmp
+unpackdep https://gmplib.org/download/gmp/gmp-6.1.2.tar.bz2 5275bb04f4863a13516b2f39392ac5e272f5e1bb8057b18aec1c9b79d73d8fb2
+cd gmp-6.1.2
+./configure --enable-static --disable-assembly --host=${target_host} CC=$CC --prefix=${LNBUILDROOT}
+make -j $num_jobs
+make install
+cd ..
+rm -rf gmp-6.1.2
+
+# download lightning
+git clone https://github.com/ElementsProject/lightning.git lightning
+cd lightning
+git checkout v0.7.3
+
+# set virtualenv for lightning
+python3 -m virtualenv venv
+. venv/bin/activate
+pip install -r requirements.txt
+
+# set standard cc for the configurator
+sed -i 's/$CC ${CWARNFLAGS-$BASE_WARNFLAGS} $CDEBUGFLAGS $COPTFLAGS -o $CONFIGURATOR $CONFIGURATOR.c/$CONFIGURATOR_CC ${CWARNFLAGS-$BASE_WARNFLAGS} $CDEBUGFLAGS $COPTFLAGS -o $CONFIGURATOR $CONFIGURATOR.c/g' configure
+sed -i 's/-Wno-maybe-uninitialized/-Wno-uninitialized/g' configure
+./configure CONFIGURATOR_CC=${CONFIGURATOR_CC} --prefix=${LNBUILDROOT} --disable-developer --disable-compat --disable-valgrind --enable-static
+
+cp /repo/lightning-gen_header_versions.h gen_header_versions.h
+# update arch based on toolchain
+sed "s'NDKCOMPILER'${CC}'" /repo/lightning-config.vars > config.vars
+sed "s'NDKCOMPILER'${CC}'" /repo/lightning-config.h > ccan/config.h
```

Comment:
Strange switch of `sed` separators. Any particular reason for not continuing to use `/` like above?

Reply:
the code used to contain slashes in sed, that was fixed but the separator wasn't changed

Reply:
ack, sed syntax could be aligned in the entire script. I don't really like using sed on pre-generated headers file, but it is a dirty trick to don't use qemu in the CI.